samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -53,28 +56,26 @@ public LocalityManager(MetadataStore metadataStore) {
   }
 
   /**
-   * Method to allow read container locality information from the {@link MetadataStore}.
-   * This method is used in {@link org.apache.samza.coordinator.JobModelManager}.
+   * Fetch the container locality information from the {@link MetadataStore}.
    *
-   * @return the map of containerId: (hostname)
+   * @return the {@code LocalityModel} for the job
    */
-  public Map<String, Map<String, String>> readContainerLocality() {
```

Comment:
Can't we have helper here something like this: `readLastSeenHostForContainer(String processorId)`

Rather than reading locality hashmap everywhere then doing a get for last seen locality

Reply:
we got rid of this code and the new method returns a model. We will need a method to return the locality of all containers or specific container & we already have a method to expose both. 

Exposing helper methods within each attributes of `ContainerLocality` like `host` or `jmx` seems too redundant up top here.

Reply:
```
Optional.ofNullable(localityManager.readLocality().getProcessorLocality(processorId))
          .map(ProcessorLocality::host)
          .orElse(null);
```

is redundant and I was refeering to make a helper method that returns this so if tomorrow there is a third place where we need last seen host we would not repeat the same thing

Reply:
We have divergence in our logic on how lack of locality is handled e.g. within `LocalityManager` vs `ContainerProcessorManager`. I see your point in case of complex logic being shared across. However, introducing a helper method to just unwrap the model seems overkill at the moment.
Also, why only host and what is special about that attribute to have a helper wrapper method and why not for jmx or jmx tunnel url? This goes back to my point of being redundant without much ROI at the cost readability because I'd need to go down one layer further to understand what `readLastSeenHostForContainer` does in its unwrapping logic.

I'd prefer to start simple and not over architect and extract/refactor as we see the need as long as the right foundations are in place for the code to evolve and extend.

Reply:
Doing same thing more than 2 times is justified enough for helper imo and not an overkill, Anyways I will live this upto your discretion